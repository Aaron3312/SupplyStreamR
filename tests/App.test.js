import React from "react";
import renderer from "react-test-renderer";
import { render, fireEvent } from "@testing-library/react-native";
import App from "../app/index"; // Asegúrate de que la ruta sea correcta

describe("App Component", () => {
	it("renders correctly", () => {
		const tree = renderer.create(<App />).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it("contains specific elements", () => {
		const { getByText, getByRole } = render(<App />);

		// Verifica la presencia de elementos de texto
		expect(getByText("SupplyStream")).toBeTruthy();

		// Verifica que el botón de login está presente y funciona
		const loginButton = getByText("Login");
		expect(loginButton).toBeTruthy();
	});

	it("changes form state on input", () => {
		const { getByPlaceholderText } = render(<App />);

		// Encuentra el campo de email y cambia su texto
		const emailInput = getByPlaceholderText("Enter your email");
		fireEvent.changeText(emailInput, "test@example.com");

		// Verifica que el texto del input ha cambiado
		expect(emailInput.props.value).toBe("test@example.com");
	});
});
