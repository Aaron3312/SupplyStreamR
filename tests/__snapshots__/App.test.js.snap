// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component renders correctly 1`] = `
<RNCSafeAreaView
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
  style={
    [
      {
        "backgroundColor": "#fffffe",
      },
      {
        "height": "100%",
      },
    ]
  }
>
  <RCTScrollView
    contentContainerStyle={
      {
        "height": "100%",
      }
    }
  >
    <View>
      <View
        style={
          [
            {
              "width": "100%",
            },
            {
              "height": "100%",
            },
            {
              "justifyContent": "center",
            },
            {
              "alignItems": "center",
            },
            {
              "paddingLeft": 16,
              "paddingRight": 16,
            },
          ]
        }
      >
        <Image
          marginBottom={-43}
          resizeMode="contain"
          source={1}
          style={
            [
              {
                "width": 234,
              },
              {
                "height": 234,
              },
            ]
          }
        />
        <View
          style={
            [
              {
                "position": "relative",
              },
              {
                "marginTop": 112,
              },
            ]
          }
        >
          <Text
            style={
              [
                {
                  "fontSize": 36,
                  "lineHeight": 40,
                },
                {
                  "fontFamily": "Poppins-Bold",
                },
                {
                  "textAlign": "center",
                },
              ]
            }
          >
            SupplyStream
          </Text>
        </View>
        <View
          style={
            [
              {
                "width": "100%",
              },
              {
                "marginTop": 28,
              },
            ]
          }
        >
          <Text
            style={
              [
                {
                  "fontSize": 16,
                  "lineHeight": 24,
                },
                {
                  "color": "#979797",
                },
                {
                  "fontFamily": "Poppins-Medium",
                },
              ]
            }
          >
            Email
          </Text>
          <View
            style={
              [
                {
                  "width": "100%",
                },
                {
                  "height": 64,
                },
                {
                  "paddingLeft": 16,
                  "paddingRight": 16,
                },
                {
                  "backgroundColor": "#d8d7d7",
                },
                {
                  "borderBottomLeftRadius": 16,
                  "borderBottomRightRadius": 16,
                  "borderTopLeftRadius": 16,
                  "borderTopRightRadius": 16,
                },
                {
                  "flexDirection": "row",
                },
                {
                  "alignItems": "center",
                },
                [
                  {
                    "marginTop": 8,
                  },
                ],
              ]
            }
          >
            <TextInput
              onChangeText={[Function]}
              placeholder="Enter your email"
              placeholderTextColor="gray-100"
              secureTextEntry={false}
              style={
                [
                  {
                    "flexBasis": "0%",
                    "flexGrow": 1,
                    "flexShrink": 1,
                  },
                  {
                    "color": "#000",
                  },
                  {
                    "fontFamily": "Poppins-SemiBold",
                  },
                  {
                    "fontSize": 16,
                    "lineHeight": 24,
                  },
                ]
              }
              value=""
            />
          </View>
        </View>
        <View
          style={
            [
              {
                "width": "100%",
              },
              {
                "marginTop": 28,
              },
            ]
          }
        >
          <Text
            style={
              [
                {
                  "fontSize": 16,
                  "lineHeight": 24,
                },
                {
                  "color": "#979797",
                },
                {
                  "fontFamily": "Poppins-Medium",
                },
              ]
            }
          >
            Password
          </Text>
          <View
            style={
              [
                {
                  "width": "100%",
                },
                {
                  "height": 64,
                },
                {
                  "paddingLeft": 16,
                  "paddingRight": 16,
                },
                {
                  "backgroundColor": "#d8d7d7",
                },
                {
                  "borderBottomLeftRadius": 16,
                  "borderBottomRightRadius": 16,
                  "borderTopLeftRadius": 16,
                  "borderTopRightRadius": 16,
                },
                {
                  "flexDirection": "row",
                },
                {
                  "alignItems": "center",
                },
                [
                  {
                    "marginTop": 8,
                  },
                ],
              ]
            }
          >
            <TextInput
              onChangeText={[Function]}
              placeholder="Enter your password"
              placeholderTextColor="gray-100"
              secureTextEntry={true}
              style={
                [
                  {
                    "flexBasis": "0%",
                    "flexGrow": 1,
                    "flexShrink": 1,
                  },
                  {
                    "color": "#000",
                  },
                  {
                    "fontFamily": "Poppins-SemiBold",
                  },
                  {
                    "fontSize": 16,
                    "lineHeight": 24,
                  },
                ]
              }
              value=""
            />
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "opacity": 1,
                }
              }
            >
              <Image
                resizeMode="contain"
                source={1}
                style={
                  [
                    {
                      "width": 36,
                    },
                    {
                      "height": 36,
                    },
                  ]
                }
              />
            </View>
          </View>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#d8d7d7",
              "borderBottomLeftRadius": 12,
              "borderBottomRightRadius": 12,
              "borderTopLeftRadius": 12,
              "borderTopRightRadius": 12,
              "justifyContent": "center",
              "marginTop": 48,
              "minHeight": 62,
              "opacity": 1,
              "width": 150,
            }
          }
        >
          <Text
            style={
              [
                {
                  "color": "#000",
                },
                {
                  "fontFamily": "Poppins-SemiBold",
                },
                {
                  "fontSize": 18,
                  "lineHeight": 28,
                },
              ]
            }
          >
            Login
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RNCSafeAreaView>
`;
